compare.go

```diff
@@ -313,25 +317,8 @@ func compareEntry(oldEntry, newEntry Entry) ([]KeyDelta, error) {
 	return results, nil
 }
 
-// Compare compares two directory hierarchy manifests, and returns the
-// list of discrepancies between the two. All of the entries in the
-// manifest are considered, with differences being generated for
-// RelativeType and FullType entries. Differences in structure (such as
-// the way /set and /unset are written) are not considered to be
-// discrepancies. The list of differences are all filesystem objects.
-//
-// keys controls which keys will be compared, but if keys is nil then all
-// possible keys will be compared between the two manifests (allowing for
-// missing entries and the like). A missing or extra key is treated as a
-// Modified type.
-//
-// If oldDh or newDh are empty, we assume they are a hierarchy that is
-// completely empty. This is purely for helping callers create synthetic
-// InodeDeltas.
-//
-// NB: The order of the parameters matters (old, new) because Extra and
-//     Missing are considered as different discrepancy types.
-func Compare(oldDh, newDh *DirectoryHierarchy, keys []Keyword) ([]InodeDelta, error) {
+// compare is the actual workhorse for Compare() and CompareSame()
+func compare(oldDh, newDh *DirectoryHierarchy, keys []Keyword, same bool) ([]InodeDelta, error) {
```

Comment:
i wonder if instead of single use bool, the argument could be on a `DifferenceType`.

Reply:
I thought about that, but I usually try to avoid functions with a boolean parameter if possible. Maybe it's a hangover from python though, I'd be happy to change it if you want.